### DISCUSSION TITLE
Query over using sensitivity Analysis

### DISCUSSION THREAD

**Discussion by aryaman3900** [biomass-dev/biomass#78] [Q&A] [answered]
Hi @himoto ,

I am doing sensitivity analysis for my EGFR model and am facing an unusual thing. I have to do parametric sensitivity analysis with two conditions: EGFR concentration 1nM and 10nM, and am doing the same by three ways:

1. In conditions (in observable.py) just entered 1nM
2. In conditions (in observable.py) just entered 10nM
3. In conditions (in observable.py) entered both 1nM and 10nM

I have attached the graphs for your reference.

1. 
![Ras](https://user-images.githubusercontent.com/88424851/141649207-71865eab-cf20-47a3-9f33-83e33a3eeba4.png)

2.
![Ras](https://user-images.githubusercontent.com/88424851/141649217-8e3f3f30-3410-44c2-96f0-86b5f34c984c.png)

3.
![Ras](https://user-images.githubusercontent.com/88424851/141649222-17e0487f-88ec-487f-838b-4384afb8156e.png)

Surprisingly the results change in attempt 3 when I enter both the concentrations simultaneously, in attempt 2 with 10nM the change for the 1st parameter was over 100 (as shown in Fig.2) but this does not happen in Fig.3.

I would like to mention that the simulation results stay consistent in all three attempts but could you please tell me why does this happen with parametric sensitivity?

**Comment by himoto** [comment, marked as answer]
Hi @aryaman3900 ,

Thank you for your inquiry.

I would like to ask you this first: can you please perform sensitivity analysis with setting options['overwrite'] to True?
```python
run_analysis(model, target='parameter', options={'overwrite': True})
```
I would like to know whether you will get the same results with the code above.

**Reply by aryaman3900** [reply]
Yes, this solves my query. Thank You very much.

**Reply by himoto** [reply]
Hi @aryaman3900 ,

Great to hear that!
Since the computation of sensitivity coefficients is time-consuming, current BioMASS tries to use the result: `sensitivity_coefficients/` if it exists. This will let users modify visualization options again and again without computing coefficients. However, it is now difficult to detect any changes in `observable.py` in a model. I think this is the cause of the problem. When you modify any code in `observable.py`, please remove the previous calculation result via setting `options={'overwrite': True})`.

**Reply by aryaman3900** [reply]
Sure, I see how BioMASS works just looking from another perspective it does seems really a nice way by the program to make the computation a little less expensive. Thanks, @himoto for explaining this to me, I will ensure to use it the correct way.

Btw just a curious observation, I noticed in the documentation that both `optimize` and `run_analysis` functions have overwrite as an option but there is no such thing in `run_simulation`. Could you please tell what is the reason for it, does the same computational cost principle doesn't applies here?

**Reply by himoto** [reply]
@aryaman3900 ,
That's a good point.
I would like to explain why  `optimize` and `run_analysis` have 'overwrite' options but `run_simulation` does not.
As you might know, the number of iteration for simulating a model is very large to estimate parameters or calculate sensitivity coefficients, and therefore time-consuming. To prevent unexpected deletion of estimation results or coefficients, these function have 'overwrite' options.
To my mind, `run_simulation` just run simulation and visualize simulation results, it does not take time and that is why I do not implement 'overwrite' option in this function.

**Reply by aryaman3900** [reply]
Sure I see. Thank You very much for clearing my query. It is very interesting to know how much has been thought at each step. Thanks!
